src/CoreFx.Private.TestUtilities/src/System/Diagnostics/RemoteExecutorTestBase.Process.cs

```diff
@@ -49,13 +50,28 @@ private static RemoteInvokeHandle RemoteInvoke(MethodInfo method, string[] args,
             }
 
             // If we need the host (if it exists), use it, otherwise target the console app directly.
-            string testConsoleAppArgs = "\"" + a.FullName + "\" " + t.FullName + " " + method.Name + " " + string.Join(" ", args);
+            var testConsoleAppArgs = new StringBuilder();
+            testConsoleAppArgs.Append(ExtraParameter);
```

Comment:
RemoteInvoke() should implement the canonical tokenizing rules - that way, tests can just pass the args they want and not worry about them coming out wrong on the other side. Perhaps there could be a "verbatim" options for tests that specifically want to pass weird command lines to test the tokenizing. We don't need to worry about that here, though.

Here's an implementation I've been using for my private projects for years:

https://github.com/AtsushiKan/test/blob/master/Paste.cs

This will also be interesting to see if Diagnostic.Process tokenizes differently on Unix systems.